### PR TITLE
fix(skore): Different y axis for comparisons in CoefficientsDisplay

### LINKED ISSUE #2393: fix(CoefficientsDisplay): If estimators have different features then `sharey` should be False
### Describe the bug

When plotting coefficients of a ComparisonReport, if the estimators have different input features, the subplots should each have their own y-axis.

### Steps/Code to Reproduce

Reproducer:
```python
from sklearn.datasets import load_iris
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import PolynomialFeatures
from sklearn.linear_model import LogisticRegression
from skore import ComparisonReport, CrossValidationReport

X, y = load_iris(return_X_y=True)
estimator_1 = LogisticRegression(random_state=42)
estimator_2 = Pipeline(
    [
        ("poly", PolynomialFeatures()),
        ("predictor", LogisticRegression(random_state=0)),
    ]
)

report = ComparisonReport(
    [CrossValidationReport(estimator_1, X, y), CrossValidationReport(estimator_2, X, y)]
)
display = report.inspection.coefficients()
display.plot()
```

### Expected Behavior

Image

but each subplot should have its own y-axis

### Actual Behavior

Image

### Environment

main (fce4f566b3ba1aeeaf94faf4fc9a490b4c199809)

### PR DESCRIPTION
**Pull request by GaetandeCast**
Closes #2393

image
image

### CODE REVIEW

**Review by auguste-probabl — APPROVED** [review]
Thanks!
